packages/app/src/debug/DebugFailedTest.vue

```diff
@@ -20,14 +26,60 @@
         {{ titlePart }}
       </span>
     </div>
+    <div
+      v-if="!props.expandable"
+      data-cy="debug-artifacts"
+      class="flex flex-grow justify-end space-x-4.5 opacity-0 test-row-artifacts pr-18px"
+    >
+      <div
+        v-for="(result, i) in debugArtifacts"
+        :key="i"
+        :data-cy="`artifact--${result.icon}`"
+      >
+        <DebugArtifacts
+          :icon="result.icon"
+          :popper-text="result.text"
+          :url="result.url"
+        />
+      </div>
+    </div>
   </div>
+  <GroupedDebugFailedTestVue
+    v-if="props.expandable"
+    :failed-tests="props.failedTestResult"
+    :groups="props.groups"
+  />
 </template>
 <script lang="ts" setup>
 import { IconChevronRightSmall } from '@cypress-design/vue-icon'
+import { SolidStatusIcon } from '@cypress-design/vue-statusicon'
+import DebugArtifacts from './DebugArtifacts.vue'
+import GroupedDebugFailedTestVue from './GroupedDebugFailedTest.vue'
+import { computed } from 'vue'
 import type { TestResults } from './DebugSpec.vue'
+import type { CloudRunGroup } from '@packages/data-context/src/gen/graphcache-config.gen'
 
 const props = defineProps<{
-  failedTestResult: TestResults
+  failedTestResult: TestResults[]
+  groups: CloudRunGroup[]
+  expandable: boolean
 }>()
 
+const debugArtifacts = computed(() => {
+  const runInstance = props.failedTestResult[0].instance
```

Comment:
I agree with this. Having all the data consolidated in a single place makes more sense.
As for the `props.failedTestResult[0]`, having made changes to the parent component to only pass props to `debugFailedTest` if there `failedTestResults` contains at least 1 entry, we can be sure here that this will be defined